core/src/main/java/org/apache/iceberg/rest/responses/PlanTableScanResponse.java

```diff
@@ -0,0 +1,127 @@
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.PlanStatus;
+import org.apache.iceberg.rest.RESTResponse;
+
+public class PlanTableScanResponse implements RESTResponse {
+  private PlanStatus planStatus;
+  private String planId;
+  private List<String> planTasks;
+  private List<FileScanTask> fileScanTasks;
+  private List<DeleteFile> deleteFiles;
+
+  public PlanTableScanResponse() {
+    // Needed for Jackson Deserialization.
+  }
+
+  public PlanTableScanResponse(
+      PlanStatus planStatus,
+      String planId,
+      List<String> planTasks,
+      List<FileScanTask> fileScanTasks,
+      List<DeleteFile> deleteFiles) {
+    this.planStatus = planStatus;
+    this.planId = planId;
+    this.planTasks = planTasks;
+    this.fileScanTasks = fileScanTasks;
+    this.deleteFiles = deleteFiles;
+  }
+
+  public PlanStatus planStatus() {
+    return planStatus;
+  }
+
+  public String planId() {
+    return planId;
+  }
+
+  public List<String> planTasks() {
+    return planTasks;
+  }
+
+  public List<FileScanTask> fileScanTasks() {
+    return fileScanTasks;
+  }
+
+  public List<DeleteFile> deleteFiles() {
+    return deleteFiles;
+  }
+
+  @Override
+  public void validate() {
+    Preconditions.checkArgument(planStatus() != null, "invalid response, status can not be null");
+    Preconditions.checkArgument(
+        planStatus() == PlanStatus.SUBMITTED && planId != null,
+        "Invalid response: planId to be non-null when status is 'submitted");
+    Preconditions.checkArgument(
+        planStatus() != PlanStatus.CANCELLED,
+        "Invalid response: 'cancelled' is not a valid status for planTableScan");
+    Preconditions.checkArgument(
+        planStatus() != PlanStatus.COMPLETED && (planTasks() != null || fileScanTasks() != null),
+        "Invalid response: tasks can only be returned in a 'completed' status");
+    Preconditions.checkArgument(
+        deleteFiles() != null && fileScanTasks() == null,
+        "Invalid response: deleteFiles should only be returned with fileScanTasks that reference them");
+  }
+
+  public static class Builder {
+    public Builder() {}
+
+    private PlanStatus planStatus;
+    private String planId;
+    private List<String> planTasks;
+    private List<FileScanTask> fileScanTasks;
+    private List<DeleteFile> deleteFiles;
+
+    public Builder withPlanStatus(PlanStatus withPlanStatus) {
+      this.planStatus = withPlanStatus;
+      return this;
+    }
+
+    public Builder withPlanId(String withPlanId) {
+      this.planId = withPlanId;
+      return this;
+    }
+
+    public Builder withPlanTasks(List<String> withPlanTasks) {
+      this.planTasks = withPlanTasks;
+      return this;
+    }
+
+    public Builder withFileScanTasks(List<FileScanTask> withFileScanTasks) {
+      this.fileScanTasks = withFileScanTasks;
+      return this;
+    }
+
+    public Builder withDeleteFiles(List<DeleteFile> withDeleteFiles) {
```

Comment:
Double check the names of these args, I think you just copied the method name

Reply:
This may have been done due to checkstyle complaining about hidden fields, my bad. It'd be better if we could have different names

Reply:
Yea this was due to checkstyle shadow/hidden field. Will use different names